src/headless_browser.ts

```diff
@@ -223,7 +231,11 @@ export class HeadlessBrowser {
     });
     this.checkForErrorResult((res as DOMOutput), command);
     // Tried and tested, and `result` is `{result: { type: "boolean", value: false } }`
-    const exists = ((res as DOMOutput).result as SuccessResult).value;
+    const exists = ((res as DOMOutput).result as SuccessResult)
+      .value as boolean;
+    if (exists !== true) {
+      await this.done();
```

Comment:
same question here about throwing an error. does this just silently close then?

Reply:
Yeah, the `done()` method just closes the ws conn and stops the subprocess - it's essentially saying "If we can see that the assertion is going to fail, cleanup Sinco"

Reply:
ohh i see `assertSee` and `assertUrlIs` are just for testing... should they even be in this class? 🤔

Reply:
Yeah that would make sense tbh, not entirely sure how to architect this but we can make an issue for it and discuss it?

Reply:
Ah just seen:

> lgtm! the testing methods still feel weird to me (in the same class as the main HeadlessBrowser) but we can address that some other time. we're so close!

Ill make an issue for it then